src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -46,6 +46,7 @@ PreReleaseRequest<I>::~PreReleaseRequest() {
   if (!m_shutting_down) {
     m_image_ctx.state->handle_prepare_lock_complete();
   }
+  delete m_on_finish;
```

Comment:
I have a little experience with coverity, but I wonder why it is complaining about this particular case. It is expected behaviour for `Context` derived classes that they delete themselves on `complete`, and we have myriads examples like this in the code.